src/index.ts

```diff
@@ -77,14 +92,40 @@ export const go = async <T, E extends Error>(
   const fn = typeof predicate === 'function' ? predicate : () => predicate;
   if (!options) return attempt(fn);
 
-  const { retries, timeoutMs } = options;
+  const { retries, timeoutMs, delay, fullTimeoutMs } = options;
+
+  let fullTimeoutExceeded = false;
+  if (fullTimeoutMs !== undefined) {
+    // Start a "full" timeout that will stop all retries after it is exceeded
+    sleep(fullTimeoutMs).then(() => {
+      fullTimeoutExceeded = true;
+    });
+  }
```

Comment:
This is a bit of a misleading concept IMO. Because based on the code you don't guarantee that the execution will timeout (stop) after `fullTimeoutMs` ms but that we won't retry after at least `fullTimeoutMs` ms. Because the flag is checked in every cycle, so it can be up to `fullTimeoutMs` + `delay`, right?

Reply:
No, `fullTimeoutMs` is exceeded when the total time (attempt times + delay times) exceeds this number. 

> you don't guarantee that the execution will timeout (stop) after fullTimeoutMs ms

It's impossible to cancel an already running promise so this is best you can do.